daemon/src/db.rs

```diff
@@ -348,6 +365,444 @@ impl Connection {
+            role: cfd.role,
+            opening_fee: cfd.opening_fee,
+            initial_funding_rate: cfd.initial_funding_rate,
+            initial_tx_fee_rate: cfd.initial_tx_fee_rate,
+            fees: cfd.fees,
+            lock_txid: cfd.lock_txid,
+            lock_dlc_vout: cfd.lock_dlc_vout,
+            settlement: Settlement::NonCollaborative {
+                commit_txid: cfd.commit_txid,
+                payout: SettlementPayout::Cet {
+                    txid: cfd.cet_txid,
+                    vout: cfd.cet_vout,
+                    payout: cfd.cet_payout,
+                    price: cfd.closing_price,
+                },
+            },
+        })
+    }
+
+    /// Load a refunded CFD from the database.
+    pub async fn load_refunded_cfd(&self, id: OrderId) -> Result<ClosedCfd> {
+        let mut conn = self.inner.acquire().await?;
+
+        let cfd = sqlx::query!(
+            r#"
+SELECT
+    uuid as "id: model::OrderId",
+    position as "position: model::Position",
+    initial_price as "initial_price: model::Price",
+    taker_leverage as "taker_leverage: model::Leverage",
+    n_contracts as "n_contracts: model::Contracts",
+    counterparty_network_identity as "counterparty_network_identity: model::Identity",
+    role as "role: model::Role",
+    opening_fee as "opening_fee: model::OpeningFee",
+    initial_funding_rate as "initial_funding_rate: model::FundingRate",
+    initial_tx_fee_rate as "initial_tx_fee_rate: model::TxFeeRate",
+    fees as "fees: model::Fees",
+    lock_txid as "lock_txid: model::Txid",
+    lock_dlc_vout as "lock_dlc_vout: model::Vout",
+    commit_txid as "commit_txid!: model::Txid",
+    refund_txid as "refund_txid!: model::Txid",
+    refund_vout as "refund_vout!: model::Vout",
+    refund_payout as "refund_payout!: model::Payout"
+FROM
+    closed_cfds
+WHERE
+    closed_cfds.id = ?
+            "#,
+            id
+        )
+        .fetch_one(&mut *conn)
+        .await?;
+
+        Ok(ClosedCfd {
+            id: cfd.id,
+            position: cfd.position,
+            initial_price: cfd.initial_price,
+            taker_leverage: cfd.taker_leverage,
+            n_contracts: cfd.n_contracts,
+            counterparty_network_identity: cfd.counterparty_network_identity,
+            role: cfd.role,
+            opening_fee: cfd.opening_fee,
+            initial_funding_rate: cfd.initial_funding_rate,
+            initial_tx_fee_rate: cfd.initial_tx_fee_rate,
+            fees: cfd.fees,
+            lock_txid: cfd.lock_txid,
+            lock_dlc_vout: cfd.lock_dlc_vout,
+            settlement: Settlement::NonCollaborative {
+                commit_txid: cfd.commit_txid,
+                payout: SettlementPayout::Refund {
+                    txid: cfd.refund_txid,
+                    vout: cfd.refund_vout,
+                    payout: cfd.refund_payout,
+                },
+            },
+        })
+    }
```

Comment:
Hmm, I am not convinced that this is a good abstraction as it is currently implemented. You don't know how a CFD was closed before you load it, hence having the caller choose the correct function is a bit weird?

Can we somehow restructure things so that we have three functions:

- `load_refunded_cfds`
- `load_force_closed_cfds`
- `load_collaboratively_closed_cfds`

Those would all return the same `ClosedCfd` which could have different constructors, based on where it is called from. Not sure if it is too wasteful but we could just reload all closed in projection upon every cycle of closing them.

I guess achieving data consistency there is actually a bit tricky. I think we want a CFD to "vanish" from the `cfds` table and appear in the `closed_cfds` atomically but at the same time, projection needs to update its local state as well.

Thinking out loud further: projection has a local cache so I think the responsibility of maintaining data consistency needs to solved there. The cache entry will still be there even if the `close_cfd::Actor` deletes all the events. Thus, `projection` should probably just regularly load all closed CFDs and update its internal cache to remove the entry and instead, populate it from the `closed_cfds` table.

Alternatively, we could make this completely transparent in the `db` layer where `db::load_cfd` can also return an instance from the `closed_cfds` table but as we discussed offline, this is a bit annoying because not every module needs to deal with closed CFDs. It does seem like a much cleaner abstraction though because it hides those DB details. Thoughts?